src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/MethodCodeNode.cs

```diff
@@ -112,10 +112,21 @@ private void AddPInvokeParameterDependencies(ref DependencyList dependencies, No
             }
             else if (MarshalHelpers.IsStructMarshallingRequired(parameter))
             {
+                var stub = (Internal.IL.Stubs.StructMarshallingThunk)factory.InteropStubManager.GetStructMarshallingManagedToNativeStub(parameter);
                 dependencies.Add(factory.ConstructedTypeSymbol(factory.InteropStubManager.GetStructMarshallingType(parameter)), "Struct Marshalling Type");
-                dependencies.Add(factory.MethodEntrypoint(factory.InteropStubManager.GetStructMarshallingManagedToNativeStub(parameter)), "Struct Marshalling stub");
+                dependencies.Add(factory.MethodEntrypoint(stub), "Struct Marshalling stub");
                 dependencies.Add(factory.MethodEntrypoint(factory.InteropStubManager.GetStructMarshallingNativeToManagedStub(parameter)), "Struct Marshalling stub");
                 dependencies.Add(factory.MethodEntrypoint(factory.InteropStubManager.GetStructMarshallingCleanupStub(parameter)), "Struct Marshalling stub");
+
+                foreach (var inlineArrayCandidate in stub.GetInilineArrayCandidates())
+                {
+                    var inlineArrayType = inlineArrayCandidate.Item1;
```

Comment:
Before we migrate over to C# 7 with named tuples, using Tuples in return is pretty ugly...